third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -0,0 +1,599 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Prerequisites: none.
+//!
+//! Simplifies the control flow graph in the following ways:
+//! (Implemented by `ControlFlowGraphSimplifier`)
+//! 1. Eliminates branch or jump to jump:
+//!     Any empty block
+//!
+//!     L1: goto L2 where L1 != L2
```

Comment:
It's easier to think about the transformations in terms of operations on the control flow graph, rather than matching and rewriting patterns of instructions. The documentation presents the transformations in the latter style only for concreteness, otherwise it's much easier to explain by pictures.
1. I’ve made the labels explicit, and other syntax is from the pretty printed stackless bytecode (as in .exp files). This may be more readable than the implementation-like syntax `Branch(L1, _else, _cond)`.
2. We don’t need to keep that `Jump(L2)` in `Label(L1); Jump(L2)` if all branches or jumps to L1 are replaced by branches or jumps to L2. Fall throughs to L1 are also documented now.
3. The title is a bit misleading. We are reducing the case where a branch or jump is followed immediately by a jump, and the second jump is eliminated. It should perhaps be “Reduces branch to jump or jump to jump”, which is unambiguous.
4. The implementation applies the described transformations one at a time, and each single application of the rule keeps the semantics of the program.